src/engine/pieces/piece.ts

```diff
@@ -17,4 +35,61 @@ export default class Piece {
         const currentSquare = board.findPiece(this);
         board.movePiece(currentSquare, newSquare);
     }
+
+    public isValidMove(board: Board, nextSquare: Square) {
+
+        let pieceOnBoard = board.getPiece(nextSquare)
+        if (pieceOnBoard === undefined)
+            return MoveDiagnostic.EMPTY_SQUARE;
+
+        if ((this.player != pieceOnBoard.player) && (pieceOnBoard.pieceType != PieceType.KING))
+            return MoveDiagnostic.CAPTURABLE_PIECE_PRESENT;
+
+        return MoveDiagnostic.UNCAPTURABLE_PIECE_PRESENT;
+    }
+
+    private getDirectionalMoves(board : Board, directions : Array<Direction>)
+    {
+        let validMoves = new Array(0);
+        let currentSquare = board.findPiece(this);
+
+        for (let direction of directions) {
+            let nextSquare = currentSquare.moveBy(direction.rowChange, direction.colChange);
+            let keepTesting = true;
+            while (board.isOnBoard(nextSquare) && keepTesting) {
+                switch (this.isValidMove(board, nextSquare)) {
+                    case MoveDiagnostic.EMPTY_SQUARE:
+                        validMoves.push(nextSquare);
+                        break;
+                    case MoveDiagnostic.CAPTURABLE_PIECE_PRESENT:
+                        validMoves.push(nextSquare);
+                        keepTesting = false;
+                        break;
+                    case MoveDiagnostic.UNCAPTURABLE_PIECE_PRESENT:
+                        keepTesting = false;
+                        break;
+                }
+                nextSquare = nextSquare.moveBy(direction.rowChange, direction.colChange);
+            }
+        }
+        return validMoves;
+    }
+
+    public getDiagonalMoves(board: Board) : Array<Square>{
+
+        let directions = [{"rowChange": 1,"colChange": 1}, 
+            {"rowChange": 1,"colChange": -1}, 
+            {"rowChange": -1,"colChange": 1}, 
+            {"rowChange": -1,"colChange": -1}];
```

Comment:
was there a reason for putting quotes around the variable names? 

Also I would maybe rename to `directionChanges` then you can name the inner values to just `row` and `col` to avoid repeating Change unnecessarily so much

Also nit pick but the convention for how this would look in code with the spacing adjusted would be 
```
     let directions = [
            {rowChange: 1, colChange: 1}, 
            {rowChange: 1, colChange: -1}, 
            {rowChange: -1, colChange: 1}, 
            {rowChange: -1, colChange: -1}
        ];
```

Reply:
same applies to the below and directions in knight